llama.cpp

```diff
@@ -886,10 +841,14 @@ struct llama_mmap {
         // prefetch/readahead impairs performance on NUMA systems
         if (numa) { prefetch = 0; }
 #ifdef __linux__
+        if (posix_fadvise64(fd, 0, file->size, POSIX_FADV_SEQUENTIAL)) {
```

Comment:
The *64 variant is a available with glibc because we define _GNU_SOURCE, but I'm not sure about Bionic (Android) or Musl.

Reply:
So I guess I need to check for `_LARGEFILE64_SOURCE`?

Reply:
Since it applies to the entire file, I think using `posix_fadvise` with 0 len should also work. The performance is the same on my system.